core/slider.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2016 Massachusetts Institute of Technology
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The slider class used for field sliders
+ * @author Chase Mortensen
+ */
+'use strict';
+
+goog.provide('Blockly.Slider');
+    
+Blockly.Slider = function() {
+
+    // Set up default values
+    this.step_ = 1;
```

Comment:
Blockly uses the prototype for setting default values. See https://github.com/microsoft/pxt-blockly/blob/develop/core/comment.js#L52